src/wallet/wallet.cpp

```diff
@@ -201,7 +203,7 @@ void CWallet::DeriveNewChildKey(const CKeyMetadata& metadata, CKey& secretRet, u
             throw std::runtime_error(std::string(__func__) + ": SetHDChain failed");
```

Comment:
`SetHDChain(hdChainCurrent, false)` should probably take in walletdb since it calls WriteHDChain, however it is also called from a function that doesn't have access to walletdb.

Reply:
This https://github.com/UdjinM6/dash/commit/523192d0e2e45c48d44db3c2daa029d589de2988 should do it. And btw, I'm seeing a huge improvement here - creating hd wallet with `keypool=10000`: before - 565s (9.5 minutes), after - 20s! Almost a 30x speedup 👍

Reply:
cherry-picked